### PR TITLE
[nrfconnect] Updated nRF Connect SDK revision

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
#### Problem
Some features like SRP requires newer OpenThread version,
which is provided by nRF Connect SDK in the nrfconnect case.

 #### Summary of Changes
* Changed used nRF Connect SDK revision.
* Updated west tool to required 0.10.1 version.
* Aligned minimal logging and PWM to the newer NCS version.
* Enabled DNS-SD SRP support by default.

### CODE REVIEW

File: scripts/requirements.in [hunk 1]
```diff
@@ -31,7 +31,7 @@ portpicker
 mobly
 
 # zephyr
-west>=0.9.0
+west>=0.10.1
```
Comment (Damian-Nordic) [inline scripts/requirements.in:34]: Update `requirements.txt` as well?

**Review by Damian-Nordic — COMMENTED** [review]
One additional cleanup which can be done is removing `CONFIG_CHIP_OPENTHREAD_CONFIG` from examples' prj.conf and setting `CONFIG_OPENTHREAD_CONFIG_NUM_MESSAGE_BUFFERS=64` in `sample-defaults.conf` as the setting can now be configured via Kconfig and the previous workaround is no longer needed.

File: scripts/requirements.in (hunk 1 above)
Comment (kkasperczyk-no) [reply scripts/requirements.in:34]: They are already updated. Michael did it some time ago.

**Review by LuDuda — APPROVED** [review]

File: .github/workflows/examples-nrfconnect.yaml [hunk 2]
```diff
@@ -40,6 +40,8 @@ jobs:
               with:
                   fetch-depth: 0
                   submodules: true
+            - name: Setup environment by running activate script.
+              run: source scripts/activate.sh
```
Comment (Damian-Nordic) [inline .github/workflows/examples-nrfconnect.yaml:44]: Looks like you need to combine it with the next step using "&&".

**Review by Damian-Nordic — APPROVED** [review]
LGTM once you solve the CI issues.

File: .github/workflows/examples-nrfconnect.yaml (hunk 2 above)
Comment (kkasperczyk-no) [reply .github/workflows/examples-nrfconnect.yaml]: Currently the problem is that scripts are ran in shell not bash, so source command is not found, but I'm working on it.

**Pull request closed without merging**